### PR TITLE
Add pinRequired configuration for gift card component

### PR DESCRIPTION
**Pull request by OscarSpruit**
## Description
Add configuration to hide the PIN input field for gift cards.

## Checklist 
- [x] Code is unit tested
- [x] Changes are tested manually
- [x] Add relevant labels to PR  

COAND-741

### CODE REVIEW

File: giftcard/src/main/java/com/adyen/checkout/giftcard/internal/ui/GiftCardDelegate.kt [hunk 1]
```diff
@@ -41,4 +41,6 @@ internal interface GiftCardDelegate :
     fun resolveBalanceResult(balanceResult: BalanceResult)
 
     fun resolveOrderResponse(orderResponse: OrderResponse)
+
+    fun isPinRequired(): Boolean
```
Comment (jreij) [inline giftcard/src/main/java/com/adyen/checkout/giftcard/internal/ui/GiftCardDelegate.kt:45]: Would this be better as part of the `GiftCardOutputData`?

Comment (OscarSpruit) [reply giftcard/src/main/java/com/adyen/checkout/giftcard/internal/ui/GiftCardDelegate.kt:45]: Hmm maybe, but then we would potentially do unnecessary initialisation when creating `GiftCardView`. I don't really have a strong preference, do you?

Comment (jreij) [reply giftcard/src/main/java/com/adyen/checkout/giftcard/internal/ui/GiftCardDelegate.kt:45]: Not really, up to you 👍

Comment (OscarSpruit) [reply giftcard/src/main/java/com/adyen/checkout/giftcard/internal/ui/GiftCardDelegate.kt:45]: Okay then I'll keep it this way

**Review by jreij — APPROVED** [review]

**Pull request merged**